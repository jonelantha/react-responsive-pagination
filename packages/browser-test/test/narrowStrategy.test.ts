import { setupThrowOnError, stringifyWithUndefined } from './helper';

beforeAll(() => {
  setupThrowOnError(page);
});

const testWidths = [150, 200, 250];

const narrowStrategies = [
  undefined,
  'dropNav',
  'dropEllipsis',
  ['dropNav', 'dropEllipsis'],
  ['dropEllipsis', 'dropNav'],
];

describe.each(narrowStrategies.map(narrowStrategy => [narrowStrategy]))(
  'Auto sizing with narrowStrategy %p',
  narrowStrategy => {
    beforeAll(async () => {
      await page.goto('about:blank');

      await page.goto(`${harnessUrl}bootstrap4`);

      await page.fill(
        '#narrowStrategyAsJson',
        stringifyWithUndefined(narrowStrategy),
      );

      await page.fill('#totalAsJson', '100');
    });

    test.each(testWidths.map(width => [width]))(
      'renders correctly with viewport width %ipx',
      async width => {
        await page.setViewportSize({ width, height: 700 });

        await page.evaluate(() => new Promise(requestAnimationFrame));

        const paginationHtml = await page.$eval('ul.pagination', ul => ul.outerHTML);

        expect(paginationHtml).toMatchSnapshot();
      },
    );
  },
);
