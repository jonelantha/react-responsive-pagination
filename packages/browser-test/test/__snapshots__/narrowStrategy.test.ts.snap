// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auto sizing with narrowStrategy "dropEllipsis" renders correctly with viewport width 150px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">1</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`Auto sizing with narrowStrategy "dropEllipsis" renders correctly with viewport width 200px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">1</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`Auto sizing with narrowStrategy "dropEllipsis" renders correctly with viewport width 250px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">1</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`Auto sizing with narrowStrategy "dropNav" renders correctly with viewport width 150px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">1</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li></ul>"`;

exports[`Auto sizing with narrowStrategy "dropNav" renders correctly with viewport width 200px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">1</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li></ul>"`;

exports[`Auto sizing with narrowStrategy "dropNav" renders correctly with viewport width 250px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">1</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`Auto sizing with narrowStrategy ["dropEllipsis", "dropNav"] renders correctly with viewport width 150px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">1</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">100</a></li></ul>"`;

exports[`Auto sizing with narrowStrategy ["dropEllipsis", "dropNav"] renders correctly with viewport width 200px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">1</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`Auto sizing with narrowStrategy ["dropEllipsis", "dropNav"] renders correctly with viewport width 250px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">1</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`Auto sizing with narrowStrategy ["dropNav", "dropEllipsis"] renders correctly with viewport width 150px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">1</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">100</a></li></ul>"`;

exports[`Auto sizing with narrowStrategy ["dropNav", "dropEllipsis"] renders correctly with viewport width 200px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">1</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li></ul>"`;

exports[`Auto sizing with narrowStrategy ["dropNav", "dropEllipsis"] renders correctly with viewport width 250px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">1</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`Auto sizing with narrowStrategy undefined renders correctly with viewport width 150px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">1</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`Auto sizing with narrowStrategy undefined renders correctly with viewport width 200px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">1</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`Auto sizing with narrowStrategy undefined renders correctly with viewport width 250px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">1</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;
